modal_utils/async_utils.py

```diff
@@ -327,7 +327,7 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> Awaitable[T]:
     return wrapper
 
 
-async def iterate_blocking(iterator: Iterator[T]) -> AsyncIterator[T]:
+async def iterate_blocking(iterator: Iterator[T]) -> AsyncGenerator[Union[T, object], None]:
```

Comment:
This return signature should not include `object` in the union - it's only used internally as a marker for when to break the loop. Better then to have a type: ignore on the yield, so at least the users of this function won't have to deal with the generic `object` case

Reply:
@freider OK looks like casting it helps avoid needing to type ignore.